controllers/machinedeploymentupgrade_controller.go

```diff
@@ -57,3 +118,92 @@ func (r *MachineDeploymentUpgradeReconciler) SetupWithManager(mgr ctrl.Manager)
 		For(&anywherev1.MachineDeploymentUpgrade{}).
 		Complete(r)
 }
+
+func (r *MachineDeploymentUpgradeReconciler) reconcile(ctx context.Context, log logr.Logger, mdUpgrade *anywherev1.MachineDeploymentUpgrade) (ctrl.Result, error) {
+	// return early if machine deployment upgrade is already complete
+	if mdUpgrade.Status.Ready {
```

Comment:
When does this get triggered to not ready if that will be determined by the status of the nodes? I guess we are assuming that when we create the objects, this field will be empty right?

Reply:
I removed this check because we should just allow this to be idempotent